Failing test: Rules Management - Prebuilt Rules (Common) Integration Tests - ESS Basic License (Air Gapped).x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/common/import_export/import_with_installing_package·ts - @ess @serverless @skipInServerlessMKI Import prebuilt rules when the package is not installed imports new prebuilt rules

A test failed on a tracked branch

```
JestAssertionError: expect(received).toEqual(expected) // deep equality

Expected: ArrayContaining [ObjectContaining {"immutable": true, "rule_id": "test-prebuilt-rule-a", "rule_source": {"is_customized": false, "type": "external"}}, ObjectContaining {"immutable": true, "rule_id": "test-prebuilt-rule-b", "rule_source": {"is_customized": true, "type": "external"}}]
Received: [{"actions": [], "author": ["Elastic"], "created_at": "2025-07-21T18:02:07.046Z", "created_by": "elastic", "description": "Mock rule A from mock 99.0.0 package", "enabled": false, "exceptions_list": [], "false_positives": [], "from": "now-30m", "id": "82eb1af8-d78e-4412-ae52-24e7095de0ac", "immutable": false, "index": ["test-*"], "interval": "5m", "language": "kuery", "license": "Elastic License v2", "max_signals": 100, "name": "Mock rule A from mock 99.0.0 package", "output_index": "", "query": "*:*", "references": [], "related_integrations": [], "required_fields": [], "revision": 0, "risk_score": 47, "risk_score_mapping": [], "rule_id": "test-prebuilt-rule-a", "rule_source": {"type": "internal"}, "setup": "", "severity": "medium", "severity_mapping": [], "tags": [], "threat": [], "to": "now", "type": "query", "updated_at": "2025-07-21T18:02:07.046Z", "updated_by": "elastic", "version": 3}, {"actions": [], "author": ["Elastic"], "created_at": "2025-07-21T18:02:07.051Z", "created_by": "elastic", "description": "Custom description", "enabled": false, "exceptions_list": [], "false_positives": [], "from": "now-30m", "id": "5a626b2f-4f99-42cf-ae42-9a07ae6aa008", "immutable": false, "index": ["test-*"], "interval": "5m", "language": "eql", "license": "Elastic License v2", "max_signals": 100, "name": "Mock rule B from mock 99.0.0 package", "output_index": "", "query": "any where true", "references": [], "related_integrations": [], "required_fields": [], "revision": 0, "risk_score": 47, "risk_score_mapping": [], "rule_id": "test-prebuilt-rule-b", "rule_source": {"type": "internal"}, "setup": "", "severity": "medium", "severity_mapping": [], "tags": ["custom-tag"], "threat": [], "to": "now", "type": "eql", "updated_at": "2025-07-21T18:02:07.051Z", "updated_by": "elastic", "version": 3}]
    at Context.<anonymous> (import_with_installing_package.ts:112:29)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at Object.apply (wrap_function.js:74:16) {
  matcherResult: {
    actual: [ [Object], [Object] ],
    expected: ArrayContaining {
      '$$typeof': Symbol(jest.asymmetricMatcher),
      sample: [Array],
      inverse: false
    },
    message: '\x1B[2mexpect(\x1B[22m\x1B[31mreceived\x1B[39m\x1B[2m).\x1B[22mtoEqual\x1B[2m(\x1B[22m\x1B[32mexpected\x1B[39m\x1B[2m) // deep equality\x1B[22m\n' +
      '\n' +
      'Expected: \x1B[32mArrayContaining [ObjectContaining {"immutable": true, "rule_id": "test-prebuilt-rule-a", "rule_source": {"is_customized": false, "type": "external"}}, ObjectContaining {"immutable": true, "rule_id": "test-prebuilt-rule-b", "rule_source": {"is_customized": true, "type": "external"}}]\x1B[39m\n' +
      'Received: \x1B[31m[{"actions": [], "author": ["Elastic"], "created_at": "2025-07-21T18:02:07.046Z", "created_by": "elastic", "description": "Mock rule A from mock 99.0.0 package", "enabled": false, "exceptions_list": [], "false_positives": [], "from": "now-30m", "id": "82eb1af8-d78e-4412-ae52-24e7095de0ac", "immutable": false, "index": ["test-*"], "interval": "5m", "language": "kuery", "license": "Elastic License v2", "max_signals": 100, "name": "Mock rule A from mock 99.0.0 package", "output_index": "", "query": "*:*", "references": [], "related_integrations": [], "required_fields": [], "revision": 0, "risk_score": 47, "risk_score_mapping": [], "rule_id": "test-prebuilt-rule-a", "rule_source": {"type": "internal"}, "setup": "", "severity": "medium", "severity_mapping": [], "tags": [], "threat": [], "to": "now", "type": "query", "updated_at": "2025-07-21T18:02:07.046Z", "updated_by": "elastic", "version": 3}, {"actions": [], "author": ["Elastic"], "created_at": "2025-07-21T18:02:07.051Z", "created_by": "elastic", "description": "Custom description", "enabled": false, "exceptions_list": [], "false_positives": [], "from": "now-30m", "id": "5a626b2f-4f99-42cf-ae42-9a07ae6aa008", "immutable": false, "index": ["test-*"], "interval": "5m", "language": "eql", "license": "Elastic License v2", "max_signals": 100, "name": "Mock rule B from mock 99.0.0 package", "output_index": "", "query": "any where true", "references": [], "related_integrations": [], "required_fields": [], "revision": 0, "risk_score": 47, "risk_score_mapping": [], "rule_id": "test-prebuilt-rule-b", "rule_source": {"type": "internal"}, "setup": "", "severity": "medium", "severity_mapping": [], "tags": ["custom-tag"], "threat": [], "to": "now", "type": "eql", "updated_at": "2025-07-21T18:02:07.051Z", "updated_by": "elastic", "version": 3}]\x1B[39m',
    name: 'toEqual',
    pass: false
  }
}
```

First failure: [kibana-on-merge - 9.1](https://buildkite.com/elastic/kibana-on-merge/builds/73370#01982e01-a642-4188-a859-0f2bfa47f24d)

<!-- kibanaCiData = {"failed-test":{"test.class":"Rules Management - Prebuilt Rules (Common) Integration Tests - ESS Basic License (Air Gapped).x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/common/import_export/import_with_installing_package·ts","test.name":"@ess @serverless @skipInServerlessMKI Import prebuilt rules when the package is not installed imports new prebuilt rules","test.failCount":14}} -->

--- a/x-pack/test/security_solution_api_integration/config/serverless/config.base.essentials.ts
+++ b/x-pack/test/security_solution_api_integration/config/serverless/config.base.essentials.ts
@@ -15,6 +15,12 @@ export interface CreateTestConfigOptions {
   junit: { reportName: string };
   kbnTestServerArgs?: string[];
   kbnTestServerEnv?: Record<string, string>;
+  /**
+   * Log message to wait for before initiating tests, defaults to waiting for Kibana status to be `available`.
+   * Note that this log message must not be filtered out by the current logging config, for example by the
+   * log level. If needed, you can adjust the logging level via `kbnTestServer.serverArgs`.
+   */
+  kbnTestServerWait?: RegExp;
 }
 
 export function createTestConfig(options: CreateTestConfigOptions) {
@@ -42,6 +48,10 @@ export function createTestConfig(options: CreateTestConfigOptions) {
         env: {
           ...svlSharedConfig.get('kbnTestServer.env'),
           ...options.kbnTestServerEnv,
+        },
+        runOptions: {
+          ...svlSharedConfig.get('kbnTestServer.runOptions'),
+          wait: options.kbnTestServerWait,
         },
       },
       testFiles: options.testFiles,

--- a/x-pack/test/security_solution_api_integration/config/serverless/config.base.ts
+++ b/x-pack/test/security_solution_api_integration/config/serverless/config.base.ts
@@ -18,6 +18,12 @@ export interface CreateTestConfigOptions {
   junit: { reportName: string };
   kbnTestServerArgs?: string[];
   kbnTestServerEnv?: Record<string, string>;
+  /**
+   * Log message to wait for before initiating tests, defaults to waiting for Kibana status to be `available`.
+   * Note that this log message must not be filtered out by the current logging config, for example by the
+   * log level. If needed, you can adjust the logging level via `kbnTestServer.serverArgs`.
+   */
+  kbnTestServerWait?: RegExp;
   suiteTags?: { include?: string[]; exclude?: string[] };
 }
 
@@ -51,6 +57,10 @@ export function createTestConfig(options: CreateTestConfigOptions) {
         env: {
           ...svlSharedConfig.get('kbnTestServer.env'),
           ...options.kbnTestServerEnv,
+        },
+        runOptions: {
+          ...svlSharedConfig.get('kbnTestServer.runOptions'),
+          wait: options.kbnTestServerWait,
         },
       },
       testFiles: options.testFiles,

--- a/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/configs/constants.ts
+++ b/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/configs/constants.ts
@@ -15,3 +15,12 @@ export const LOGGING_CONFIG = [
     level: 'debug',
   },
 ];
+
+/**
+ * A log message indicating that Fleet plugin has completed any necessary setup logic
+ * to make sure test suites can run without race conditions with Fleet plugin initialization.
+ *
+ * The message must not be filtered out by the logging configuration. Subsequently higher log level is better.
+ * "Fleet setup completed" has the same "info" level as "Kibana server is ready" log message.
+ */
+export const FLEET_PLUGIN_READY_LOG_MESSAGE_REGEXP = /Fleet setup completed/;

--- a/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/configs/ess/rules_management.basic.config.ts
+++ b/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/configs/ess/rules_management.basic.config.ts
@@ -6,7 +6,7 @@
  */
 
 import { FtrConfigProviderContext } from '@kbn/test';
-import { LOGGING_CONFIG } from '../constants';
+import { LOGGING_CONFIG, FLEET_PLUGIN_READY_LOG_MESSAGE_REGEXP } from '../constants';
 
 export default async function ({ readConfigFile }: FtrConfigProviderContext) {
   const functionalConfig = await readConfigFile(
@@ -21,6 +21,9 @@ export default async function ({ readConfigFile }: FtrConfigProviderContext) {
         ...functionalConfig.get('kbnTestServer.serverArgs'),
         `--logging.loggers=${JSON.stringify(LOGGING_CONFIG)}`,
       ],
+      runOptions: {
+        wait: FLEET_PLUGIN_READY_LOG_MESSAGE_REGEXP,
+      },
     },
   };
 }

--- a/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/configs/ess/rules_management.trial.config.ts
+++ b/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/configs/ess/rules_management.trial.config.ts
@@ -6,7 +6,7 @@
  */
 
 import { FtrConfigProviderContext } from '@kbn/test';
-import { LOGGING_CONFIG } from '../constants';
+import { LOGGING_CONFIG, FLEET_PLUGIN_READY_LOG_MESSAGE_REGEXP } from '../constants';
 
 export default async function ({ readConfigFile }: FtrConfigProviderContext) {
   const functionalConfig = await readConfigFile(
@@ -21,6 +21,9 @@ export default async function ({ readConfigFile }: FtrConfigProviderContext) {
         ...functionalConfig.get('kbnTestServer.serverArgs'),
         `--logging.loggers=${JSON.stringify(LOGGING_CONFIG)}`,
       ],
+      runOptions: {
+        wait: FLEET_PLUGIN_READY_LOG_MESSAGE_REGEXP,
+      },
     },
   };
 }

--- a/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/configs/serverless/rules_management.complete.config.ts
+++ b/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/configs/serverless/rules_management.complete.config.ts
@@ -7,11 +7,12 @@
 
 import type { CreateTestConfigOptions } from '../../../../../config/serverless/config.base';
 import { createTestConfig } from '../../../../../config/serverless/config.base';
-import { LOGGING_CONFIG } from '../constants';
+import { LOGGING_CONFIG, FLEET_PLUGIN_READY_LOG_MESSAGE_REGEXP } from '../constants';
 
 export function createCompleteTierTestConfig(options: CreateTestConfigOptions) {
   return createTestConfig({
     kbnTestServerArgs: [`--logging.loggers=${JSON.stringify(LOGGING_CONFIG)}`],
+    kbnTestServerWait: FLEET_PLUGIN_READY_LOG_MESSAGE_REGEXP,
     ...options,
   });
 }

--- a/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/configs/serverless/rules_management.essentials.config.ts
+++ b/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/configs/serverless/rules_management.essentials.config.ts
@@ -7,11 +7,12 @@
 
 import type { CreateTestConfigOptions } from '../../../../../config/serverless/config.base.essentials';
 import { createTestConfig } from '../../../../../config/serverless/config.base.essentials';
-import { LOGGING_CONFIG } from '../constants';
+import { LOGGING_CONFIG, FLEET_PLUGIN_READY_LOG_MESSAGE_REGEXP } from '../constants';
 
 export function createEssentialsTierTestConfig(options: CreateTestConfigOptions) {
   return createTestConfig({
     kbnTestServerArgs: [`--logging.loggers=${JSON.stringify(LOGGING_CONFIG)}`],
+    kbnTestServerWait: FLEET_PLUGIN_READY_LOG_MESSAGE_REGEXP,
     ...options,
   });
 }

--- a/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/common/import_export/import_with_installing_package.ts
+++ b/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/common/import_export/import_with_installing_package.ts
@@ -68,8 +68,7 @@ export default ({ getService }: FtrProviderContext): void => {
     },
   };
 
-  // FLAKY: https://github.com/elastic/kibana/issues/228844
-  describe.skip('@ess @serverless @skipInServerlessMKI Import prebuilt rules when the package is not installed', () => {
+  describe('@ess @serverless @skipInServerlessMKI Import prebuilt rules when the package is not installed', () => {
     beforeEach(async () => {
       await deletePrebuiltRulesFleetPackage({ supertest, es, log, retryService });
       await deleteAllRules(supertest, log);

--- a/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/common/prebuilt_rules_package/air_gapped/bootstrap_prebuilt_rules.ts
+++ b/x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/common/prebuilt_rules_package/air_gapped/bootstrap_prebuilt_rules.ts
@@ -20,8 +20,7 @@ export default ({ getService }: FtrProviderContext): void => {
   const retryService = getService('retry');
   const securitySolutionApi = getService('securitySolutionApi');
 
-  // Failing: See https://github.com/elastic/kibana/issues/229297
-  describe.skip('@ess @serverless @skipInServerlessMKI Bootstrap Prebuilt Rules', () => {
+  describe('@ess @serverless @skipInServerlessMKI Bootstrap Prebuilt Rules', () => {
     beforeEach(async () => {
       await deletePrebuiltRulesFleetPackage({ supertest, es, log, retryService });
       await deleteEndpointFleetPackage({ supertest, es, log, retryService });

